doc/design/record_fetcher_proposal.md

```diff
@@ -37,19 +38,18 @@ and persist it in a meaningful way. Analysis of this information is not in scope
 
 A manifest needs to be able to represent the raw materials associated with a record. The minimal identifiers for the
 record itself are:
-* a unique ID
+* a UUIDv4
 * which region and record type the manifest represents (e.g., "USA/FL/Bay/Court")
 * some sort of case identifier
 * the timestamp of collection
 
 A bunch of record-identifying metadata is cool, but we also need to be able to represent the documents that compose the
 record. To that end, we'll require a list of document references. Each reference will contain:
+* a label for the document
 * a content type (we can just use [MIME
 types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) here)
-* a digest of the content
-* a URI for where that document's contents are durably stored in PDAP infrastructure
-  * this isn't necessary if we adopt storage infrastructure that allows us to reliably derive this from the digest
-* an optional comment expressing where this specific document came from
+* a digest of the content, which may be used to later fetch the document from PDAP-owned storage
+* an optional comment, potentially expressing where this specific document came from
 
 In order for us to trust our collected record results, we need to be able to verify that we trust the software which
 performed the collection. This means being able to audit the code that created each manifest. To this end we also need
```

Comment:
Just for clarity, is the plan not to run all scrapers in the cloud under our control and to also generate the docker images we use directly from the source in our own build process?

I've had to be away from this for a bit, so I've without a doubt missed details as they developed.

Reply:
That is still the plan. This design doesn't work in an environment where we cannot trust the infrastructure.

Something that may have changed is that we're moving to a model where we split raw record collection (in the form of multiple documents) and the extraction of information from those documents. This document strives to:
1. define a mechanism for representing the relationships between those raw documents
2. make any tampering evident
3. leave us something we can latch on to in order to verify that all data came from PDAP code and infra

Reply:
Can't 3 be ensured through some combination of mtls and by not having the message queue / upload interface be available beyond the cloud network? What would the 'story' of tampering look like if everything was internal to our network, other than the scrape target? (aka the court record website)

Reply:
I'm not really sure how to respond to James here other than really just mitigation of potential source overload.

If we were to blame for a denial of service attack because someone was very aggressively scraping data and funneling it to us it could look bad.

Reply:
My disconnect may just come from my (erroneous?) assumption that people will contribute scrapers, but they will be executed from within our control after being audited (pr review). No bad actor could funnel data (good or bad) to us in that model.